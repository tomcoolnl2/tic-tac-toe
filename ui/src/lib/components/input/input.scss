
.form-input {
    border: 2px solid var(--background-color-secondary);
    border-radius: 4px;
    position: relative;

    &.icon {
        border-left-width: 36px;

        i {
            position: absolute;
            top: 50%;
            left: -24px;
            transform: translateY(-50%);
        }
    }

}

input {
    outline: none;
    background-color: var(--light-secondary);
    color: white;
    padding: 10px;
    width: 100%;
    opacity: 0.8;
    border-radius: 4px;

    &:disabled {
        cursor: not-allowed;
    }
}

[data-lastpass-icon-root] {
    display: none !important;
}
