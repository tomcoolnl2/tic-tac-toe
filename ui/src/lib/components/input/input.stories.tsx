import type { Meta, StoryFn } from '@storybook/react';
import { type ContainerProps, withContainer } from '../../../../.storybook/decorators';
import { type Props, Input } from './input';

const EnchantedInput: React.FC<Props & ContainerProps> = withContainer(Input, 300);

const Story: Meta<typeof Input> = {
	component: Input,
	title: 'Components/Input',
};
export default Story;

const Template: StoryFn<Props> = (args) => <EnchantedInput {...args} />;

export const TextInput = Template.bind({});
TextInput.args = {
	icon: 'user',
	id: 'username',
	value: 'test',
} as Props;

export const PasswordInput = Template.bind({});
PasswordInput.args = {
	icon: 'lock',
	type: 'password',
	id: 'password',
	value: '',
} as Props;
