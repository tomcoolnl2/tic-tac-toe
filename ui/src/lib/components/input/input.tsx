import './input.scss';

export interface Props extends React.InputHTMLAttributes<HTMLInputElement> {
	name?: string;
}

export const Input: React.FC<Props> = ({
	type = 'text',
	id,
	name,
	disabled,
	value,
	onChange,
}) => {
	return (
		<input
			type={type}
			id={id}
			name={name ?? id}
			disabled={disabled || false}
			value={value}
			autoComplete="off"
			onChange={onChange}
		/>
	);
};
