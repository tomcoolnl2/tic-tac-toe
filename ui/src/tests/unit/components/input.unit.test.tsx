import '@testing-library/jest-dom';
import { render, fireEvent } from '@testing-library/react';
import { Input, type Props } from '../../../lib/components/input/input';

describe('Input component', () => {
	const defaultProps: Props = {
		onChange: jest.fn(),
		id: 'test-input',
		value: '',
		testId: 'test-input',
	};

	it('renders without crashing', () => {
		const { container } = render(<Input {...defaultProps} />);
		expect(container.firstChild).toMatchSnapshot();
	});

	it('renders with icon when provided', () => {
		const { container } = render(<Input {...defaultProps} icon="search" />);
		expect(container.querySelector('.icon-search')).toBeTruthy();
	});

	it('calls onChange callback when input value changes', () => {
		const onChange = jest.fn();
		const { getByTestId } = render(<Input {...defaultProps} onChange={onChange} />);
		const input = getByTestId('test-input') as HTMLInputElement;

		fireEvent.change(input, { target: { value: 'test' } });
		expect(onChange).toHaveBeenCalledTimes(1);
	});

	it('sets default name as id if name is not provided', () => {
		const { getByTestId } = render(<Input {...defaultProps} />);
		const input = getByTestId('test-input') as HTMLInputElement;

		expect(input.getAttribute('name')).toBe('test-input');
	});

	it('disables input when disabled prop is true', () => {
		const { getByTestId } = render(<Input {...defaultProps} disabled />);
		const input = getByTestId('test-input') as HTMLInputElement;

		expect(input.disabled).toBe(true);
	});

	// Add more test cases as needed based on specific functionality
});
