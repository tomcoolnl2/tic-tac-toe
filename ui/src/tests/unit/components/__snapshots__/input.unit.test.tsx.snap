// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders without crashing 1`] = `
<div
  class="form-input"
>
  <i
    class="icon-undefined"
  />
  <input
    autocomplete="off"
    data-testid="test-input"
    id="test-input"
    name="test-input"
    type="text"
    value=""
  />
</div>
`;
